### PR TITLE
fastp: init 0.20.1

### PR DESCRIPTION
**Pull request by jbedo**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/fastp/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, fetchFromGitHub
+, zlib 
+}:
+
+stdenv.mkDerivation rec {
+  name = "fastp-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/fastp/default.nix:7]:
```suggestion
  pname = "fast";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/fastp/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "sha256-pANwppkO9pfV9vctB7HmNCzYRtf+Xt+5HMKzvFuvyFM=";
+  };
+
+  buildInputs = [ zlib ];
+
+  installPhase = ''
+    install -D fastp $out/bin/fastp
+  '';
+
+  meta = with stdlib.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/fastp/default.nix:23]:
```suggestion
  meta = with stdenv.lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please fix the editor config check.

File: pkgs/applications/science/biology/fastp/default.nix (hunk 2 above)
Comment (jbedo) [reply pkgs/applications/science/biology/fastp/default.nix]: Apologies!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108432` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>fastp</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108432` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>fastp</li>
</ul>
</details>

**Pull request merged**